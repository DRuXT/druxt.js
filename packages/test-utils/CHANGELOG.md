# druxt-test-utils

## 0.2.15

### Patch Changes

- Updated dependencies [41cab3a0]
  - druxt@0.22.0

## 0.2.14

### Patch Changes

- Updated dependencies [e46a3290]
  - druxt@0.21.0

## 0.2.13

### Patch Changes

- Updated dependencies [49b67872]
  - druxt@0.20.0

## 0.2.12

### Patch Changes

- Updated dependencies [be21952]
  - druxt@0.19.0

## 0.2.11

### Patch Changes

- Updated dependencies [e6cf1fc]
  - druxt@0.18.3

## 0.2.10

### Patch Changes

- 540afca: Updated drupal-jsonapi-params to 2.0.0
- Updated dependencies [540afca]
  - druxt@0.18.2

## 0.2.9

### Patch Changes

- Updated dependencies [80164a1]
  - druxt@0.18.1

## 0.2.8

### Patch Changes

- Updated dependencies [e3d5238]
  - druxt@0.18.0

## 0.2.7

### Patch Changes

- Updated dependencies [b79701c]
  - druxt@0.17.0

## 0.2.6

### Patch Changes

- Updated dependencies [7b749bd]
  - druxt@0.16.0

## 0.2.5

### Patch Changes

- Updated dependencies [2ae1d6d]
  - druxt@0.15.0

## 0.2.4

### Patch Changes

- Updated dependencies [45bc0b9]
  - druxt@0.14.0

## 0.2.3

### Patch Changes

- Updated dependencies [77ab204]
  - druxt@0.13.0

## 0.2.2

### Patch Changes

- Updated dependencies [4504a2f]
  - druxt@0.12.0

## 0.2.1

### Patch Changes

- Updated dependencies [75ff8a9]
  - druxt@0.11.0

## 0.2.0

### Patch Changes

- Updated dependencies [715e5ef]
  - druxt@0.10.0

## 0.1.0

### Patch Changes

- Updated dependencies [21170fb]
  - druxt@0.9.0
