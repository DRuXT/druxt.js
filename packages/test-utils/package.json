{
  "name": "druxt-test-utils",
  "version": "0.1.0",
  "description": "Test utilities for Druxt development.",
  "keywords": [
    "druxt",
    "jsonapi",
    "test"
  ],
  "exports": {
    ".": {
      "require": "./dist/druxt-test-utils.ssr.js",
      "import": "./dist/druxt-test-utils.esm.js"
    }
  },
  "main": "dist/druxt-test-utils.ssr.js",
  "module": "dist/druxt-test-utils.esm.js",
  "dependencies": {
    "drupal-jsonapi-params": "^1.2.2",
    "jest-mock-axios": "^4.4.0"
  },
  "peerDependencies": {
    "druxt": "*"
  }
}
