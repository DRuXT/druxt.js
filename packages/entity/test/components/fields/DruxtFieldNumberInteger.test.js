import { createLocalVue, shallowMount } from '@vue/test-utils'
import Vuex from 'vuex'
import mockAxios from 'jest-mock-axios'

import DruxtFieldNumberInteger from '../../../src/components/fields/DruxtFieldNumberInteger.vue'

jest.mock('axios')

// Setup local vue instance.
const localVue = createLocalVue()
localVue.use(Vuex)

const stubs = ['nuxt-link']

const mountComponent = (prefix = false, suffix = false, options) => {
  const propsData = {
    value: [0, 1, 2],
    schema: {
      settings: {
        config: {
          prefix,
          suffix
        },
        display: {
          prefix_suffix: !!(prefix || suffix)
        }
      }
    }
  }

  return shallowMount(DruxtFieldNumberInteger, { ...options, localVue, propsData, stubs })
}

describe('Component - DruxtFieldNumberInteger', () => {
  beforeEach(() => {
    mockAxios.reset()
  })

  test('default', async () => {
    const wrapper = mountComponent()

    expect(wrapper.vm.prefix).toBe(false)
    expect(wrapper.vm.suffix).toBe(false)

    expect(wrapper.vm.items.length).toBe(3)
    expect(wrapper.html()).toMatchSnapshot()
  })

  test('prefix', async () => {
    const wrapper = mountComponent('#')

    expect(wrapper.vm.prefix).toBe('#')
    expect(wrapper.vm.suffix).toBe(false)

    expect(wrapper.vm.items.length).toBe(3)
    expect(wrapper.html()).toMatchSnapshot()
  })

  test('suffix', async () => {
    const wrapper = mountComponent(false, ' item(s)')

    expect(wrapper.vm.prefix).toBe(false)
    expect(wrapper.vm.suffix).toBe(' item(s)')

    expect(wrapper.vm.items.length).toBe(3)
    expect(wrapper.html()).toMatchSnapshot()
  })

  test('prefix_suffix', async () => {
    const wrapper = mountComponent('AUD $', '.00')

    expect(wrapper.vm.prefix).toBe('AUD $')
    expect(wrapper.vm.suffix).toBe('.00')

    expect(wrapper.vm.items.length).toBe(3)
    expect(wrapper.html()).toMatchSnapshot()
  })
})
