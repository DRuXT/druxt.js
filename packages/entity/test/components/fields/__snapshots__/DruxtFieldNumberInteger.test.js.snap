// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component - DruxtFieldNumberInteger default 1`] = `
"<div>
  <!---->
  <!----> <span><!---->
    0
    <!----></span><span><!---->
    1
    <!----></span><span><!---->
    2
    <!----></span></div>"
`;

exports[`Component - DruxtFieldNumberInteger prefix 1`] = `
"<div>
  <!---->
  <!----> <span><span class=\\"prefix\\">#</span>
  0
  <!----></span><span><span class=\\"prefix\\">#</span>
  1
  <!----></span><span><span class=\\"prefix\\">#</span>
  2
  <!----></span></div>"
`;

exports[`Component - DruxtFieldNumberInteger prefix_suffix 1`] = `
"<div>
  <!---->
  <!----> <span><span class=\\"prefix\\">AUD $</span>
  0
  <span class=\\"suffix\\">.00</span></span><span><span class=\\"prefix\\">AUD $</span>
  1
  <span class=\\"suffix\\">.00</span></span><span><span class=\\"prefix\\">AUD $</span>
  2
  <span class=\\"suffix\\">.00</span></span></div>"
`;

exports[`Component - DruxtFieldNumberInteger suffix 1`] = `
"<div>
  <!---->
  <!----> <span><!---->
    0
    <span class=\\"suffix\\"> item(s)</span></span><span><!---->
    1
    <span class=\\"suffix\\"> item(s)</span></span><span><!---->
    2
    <span class=\\"suffix\\"> item(s)</span></span></div>"
`;
